install/kubernetes/cilium/charts/config/templates/configmap.yaml

```diff
@@ -477,15 +477,19 @@ data:
 {{- if and .Values.global.sockops .Values.global.sockops.enabled }}
   sockops-enable: {{ .Values.global.sockops.enabled | quote }}
 {{- end }}
-{{- if and .Values.global.k8s .Values.global.k8s.requireIPv4PodCIDR }}
+{{- if and (hasKey .Values.global "k8s") (hasKey .Values.global.k8s "requireIPv4PodCIDR") }}
   k8s-require-ipv4-pod-cidr: {{ .Values.global.k8s.requireIPv4PodCIDR | quote }}
 {{- else }}
 {{- if eq $ipam "cluster-pool" }}
```

Comment:
Correct me if I'm wrong but I believe we should be checking for this condition first? Otherwise, regardless of the IPAM being `"cluster-pool"` or not, the `{- if and (hasKey .Values.global "k8s") (hasKey .Values.global.k8s "requireIPv4PodCIDR") }}` will always take priority.

Reply:
Correct, what should be done, if `$ipam` is set to `cluster-pool`, then `k8s-require-ipv4-pod-cidr` always true?